driver.go

```diff
@@ -107,11 +107,16 @@ func extractConnectorParams(paramsString string) (map[string]string, error) {
 	}
 	keyValuePairs := strings.Split(paramsString, ";")
 	for _, keyValueString := range keyValuePairs {
+		if keyValueString == "" {
+			// Ignore empty parameter entries in the string, for example if
+			// the connection string contains a trailing ';'.
+			continue
+		}
 		keyValue := strings.SplitN(keyValueString, "=", 2)
 		if keyValue == nil || len(keyValue) != 2 {
 			return nil, fmt.Errorf("invalid connection property: %s", keyValueString)
 		}
-		params[keyValue[0]] = keyValue[1]
+		params[strings.ToLower(keyValue[0])] = keyValue[1]
```

Comment:
why is this case insensitive?

Reply:
This makes sure that any additional parameters in the connection string are case insensitive. So a user does not have to worry whether the parameter name is for example `usePlainText` or `useplaintext` (that parameter name would normally be clear for everyone, but for parameters like `readonly` vs `readOnly` and `autocommit` vs `autoCommit` it's easy to make a mistake).